yucca/training/data_loading/YuccaDataset.py

```diff
@@ -16,19 +17,30 @@ def __init__(
         self,
         preprocessed_data_dir: list,
         patch_size: list | tuple,
-        keep_in_ram=False,
-        seg_dtype: type = int,
+        keep_in_ram: Union[bool, None] = None,
+        label_dtype: type = int,
         composed_transforms=None,
     ):
         self.all_cases = preprocessed_data_dir
         self.composed_transforms = composed_transforms
-        self.keep_in_ram = keep_in_ram
         self.patch_size = patch_size
-        self.seg_dtype = seg_dtype
+        self.label_dtype = label_dtype
 
         self.already_loaded_cases = {}
         self.croppad = CropPad(patch_size=self.patch_size, p_oversample_foreground=0.33)
-        self.to_torch = NumpyToTorch(seg_dtype=self.seg_dtype)
+        self.to_torch = NumpyToTorch(label_dtype=self.label_dtype)
+
+        self._keep_in_ram = keep_in_ram
+
+    @property
+    def keep_in_ram(self):
+        if self._keep_in_ram is not None:
+            return self._keep_in_ram
+        if len(self.all_cases) < 1000:
```

Comment:
1000 is rather arbitrary as it will also depend on the hardware, but with this limit we're erring on the conservative side to not run into OOM errors during training